mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/nocompatibility/InboundToAttributesTranslator.java

```diff
@@ -1,43 +1,149 @@
+ * Translates between mule 3 inbound properties to mule 4 attributes
+ *
+ * @author Mulesoft Inc.
+ * @since 1.3.0
+ */
 public class InboundToAttributesTranslator {
-  
-  private static List<String> SUPPORTED_SOURCES = ImmutableList.of("http:listener", "file:listener", "http:request");
 
-  public static String translate(String originatingSource, String propertyToTranslate) {
+  public static final SourceType HTTP_LISTENER =
+      new SourceType(HTTP_NAMESPACE_URI, "listener");
+  public static final SourceType HTTP_TRANSPORT =
+      new SourceType(HTTP_NAMESPACE_URI, "inbound-endpoint");
+  public static final SourceType HTTP_CONNECTOR_REQUESTER = new SourceType(HTTP_NAMESPACE_URI, "request");
+  public static final SourceType HTTP_TRANSPORT_OUTBOUND = new SourceType(HTTP_NAMESPACE_URI, "outbound-endpoint");
+  public static final SourceType HTTP_POLLING_CONNECTOR = new SourceType(HTTP_NAMESPACE_URI, "polling-connector");
+  public static final SourceType FILE_INBOUND = new SourceType(FILE_NAMESPACE_URI, "listener");
+  public static final SourceType IMAP_INBOUND = new SourceType(IMAP_NAMESPACE_URI, "inbound-endpoint");
+  public static final SourceType POP3_INBOUND = new SourceType(POP3_NAMESPACE_URI, "inbound-endpoint");
+  public static final SourceType FTP_INBOUND = new SourceType(FTP_NAMESPACE_URI, "inbound-endpoint");
+  public static final SourceType JMS_INBOUND = new SourceType(JMS_NAMESPACE_URI, "inbound-endpoint");
+  public static final SourceType JMS_OUTBOUND = new SourceType(JMS_NAMESPACE_URI, "outbound-endpoint");
+  public static final SourceType REQUEST_REPLY =
+      new SourceType(CORE_NS_URI, "request-reply");
+  public static final SourceType QUARTZ_INBOUND =
+      new SourceType(QUARTZ_NS_URI, "inbound-endpoint");
+  public static final SourceType SFTP_INBOUND = new SourceType(SFTP_NS_URI, "inbound-endpoint");
+  public static final SourceType WS_CONSUMER = new SourceType(WS_NAMESPACE_URI, "consumer");
+  private static String ATTRIBUTES_PATTERN_REGEX = "[a-zA-Z0-9_\\-.]*";
+
+  private static Map<SourceType, Class> translatorClasses;
+
+  public InboundToAttributesTranslator() {
+    translatorClasses = new ImmutableMap.Builder<SourceType, Class>()
+        .put(HTTP_LISTENER, HttpConnectorListener.class)
+        .put(HTTP_TRANSPORT, HttpConnectorListener.class)
+        .put(HTTP_CONNECTOR_REQUESTER, HttpConnectorRequester.class)
+        .put(HTTP_TRANSPORT_OUTBOUND, HttpConnectorRequester.class)
+        .put(HTTP_POLLING_CONNECTOR, HttpConnectorRequester.class)
+        .put(FILE_INBOUND, FileInboundEndpoint.class)
+        .put(IMAP_INBOUND, AbstractEmailSourceMigrator.class)
+        .put(POP3_INBOUND, AbstractEmailSourceMigrator.class)
+        .put(FTP_INBOUND, FtpInboundEndpoint.class)
+        .put(JMS_INBOUND, AbstractJmsEndpoint.class)
+        .put(JMS_OUTBOUND, AbstractJmsEndpoint.class)
+        .put(REQUEST_REPLY, AbstractJmsEndpoint.class)
+        .put(QUARTZ_INBOUND, AbstractJmsEndpoint.class)
```

Comment:
I think quartz may reference any type of inbound endpoint, not only JMS
Not sure about REQUEST_REPLY, buy may also be able to use VM besides JMS

Reply:
I based these translations on what we have today in MMA, quartz is only supported with JMS, so if we want to preserve the current behavior I think it's safe for now to do this. Same for "request-reply". In the future we need to think how to add more endpoints for quartz then